pkg/agent/openflow/pipeline.go

```diff
@@ -3091,7 +3121,7 @@ func (f *featureService) gatewaySNATFlows() []binding.Flow {
 			MatchCTStateTrk(true).
 			MatchRegMark(FromGatewayRegMark, ToGatewayRegMark).
 			Action().CT(true, SNATMarkTable.GetNext(), f.dnatCtZones[ipProtocol], f.ctZoneSrcField).
-			LoadToCtMark(ConnSNATCTMark, HairpinCTMark).
+			LoadToCtMark(ConnSNATCTMark, HairpinCTMark, ConnAllowedCTMark).
```

Comment:
One more question, if the `ConnAllowedCTMark` is added in `SNATMarkTable` and committed to the dnatZone, but the packets is dropped by an ingress rule, will it cause a wrong result?

Reply:
That's not possible for a SNAT connection to be dropped by an ingress rule AFAIK:
- hairpin connections skip ingress policy enforcement
- other connections for which we do SNAT are not destined to a local Pod

I think I have a quick note about this in the OVS pipeline doc.